database/redis/metric.go

```diff
@@ -15,8 +15,15 @@ import (
 
 // GetPatterns gets updated patterns array
 func (connector *DbConnector) GetPatterns() ([]string, error) {
-	c := connector.pool.Get()
+	var pool *redis.Pool
```

Comment:
I'm not sure this is the right place to decide. Some callers may want to use only strictly consistent master database, some may be satisfied with stale, eventually-consistent data. You propose to use slave reads for all `GetPatterns` callers, now and in the future. This can be unexpected to developers and lead to elusive bugs.

I see two possible solutions:
1. Pass a special parameter (`GetPatterns(allowStale bool)`?) to this function that allows slave reads.
2. Create a copy of this function that works with slaves (`GetPatternsAllowStale`?).
3. Create a second `DbConnector`, that uses slaves for all read (and write, if possible?) operations.

Reply:
Yeah, I did not really mean to leave it at that. It's a fast implementation to see if it does any difference, and if it proves useful, refactoring and rethinking would be in order.